extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoP.java

```diff
@@ -383,20 +400,29 @@ boolean everCompletes() {
         @Nonnull
         @Override
         public Traverser<?> nextChunkTraverser() {
-            Traverser<ChangeStreamDocument<Document>> traverser = cursor::tryNext;
-            return traverser.flatMap(doc -> {
-                resumeToken = doc.getResumeToken();
-                long eventTime = clusterTime(doc);
-                I item = mapFn.apply(doc);
-                return item == null
-                        ? Traversers.empty()
-                        : eventTimeMapper.flatMapEvent(item, 0, eventTime);
-            });
+            try {
+                List<ChangeStreamDocument<Document>> chunk = readChunk(cursor, BATCH_SIZE);
+                if (chunk.isEmpty()) {
+                    return eventTimeMapper.flatMapIdle();
+                }
+
+                return traverseIterable(chunk)
+                        .flatMap(doc -> {
+                            resumeToken = doc.getResumeToken();
+                            long eventTime = clusterTime(doc);
+                            I item = mapFn.apply(doc);
+                            return item == null
+                                    ? eventTimeMapper.flatMapIdle()
+                                    : eventTimeMapper.flatMapEvent(item, 0, eventTime);
```

Comment:
Will it ever get null here?

Reply:
You mean `item == null`? Yes, if mapFn returns null. EventTimeMapper does not provide simple `map`